### ISSUE TITLE
[FEATURE] Add New Disks via Config of Longhorn / HELM

### ISSUE THREAD

**Issue by Rohmilchkaese** [longhorn/longhorn#8498]
## Is your feature request related to a problem? Please describe (👍 if you like this request)

It would be neat to be able to add new disks not via ui but via config of values from eg. Helm

## Describe the solution you'd like

Make Disks that longhorn uses more precise. Eg. not only stadnard storage path but being able to add new disks to longhorn via config as well

**Comment by ejweber** [comment]
Yes, I think this fits the GitOps mindset well. How does it work in practice, though?

- Do we assume a homogeneous cluster made up of nodes that all have disks mounted to the same locations? Then we can add some kind of disk config section that applies to all nodes.
- If the cluster is not homogeneous, is the disk config best effort? For example, create a disk from this path IF IT EXISTS.
- Or do we need to allow for a large configuration structure that maps node names (hopefully not) or node labels/annotations (better) to disk config structures?

From an implementation standpoint, maybe we'd need to use an intermediary ConfigMap like we do for default settings today (though I guess we could just store the whole structure in a setting itself?).

**Comment by PhanLe1010** [comment]
Adding this to the backlog so @derekbit can assign it later

**Comment by thecodeassassin** [comment]
Just want to tag along here:

https://github.com/longhorn/longhorn-manager/blob/master/types/types.go#L896

It would be great if one could specify the disk type when configuring default disks using annotations.
```
labels:
    node.longhorn.io/create-default-disk: "config"
},
annotations:
    node.longhorn.io/default-disks-config:
        '[{"path":"/dev/nvme2n1","allowScheduling":true}
```        
This doesn't work currently because the config always assumes the disk type is FIle System.

 this  way one could prep nodes before installing longhorn and use dedicated disks.
 
 I am thinking about adding this feature to instance manager myself, looks fairly doable. if v2 engine support is enabled it's just a matter of adding disks anyways.

**Comment by derekbit** [comment]
Let's address it in v1.8.0.

**Comment by Rohmilchkaese** [comment]
Hey team, just wanted to follow up on this since it’s been a while.
This was originally targeted for v1.8.0 back in June 2024, then moved to v1.9.0, then v1.10.0, and now it’s back in the generic Backlog with no milestone. I understand priorities shift, but this has been bumped three times now and I’d appreciate some transparency on whether this is still planned.
The core problem remains: without the ability to configure disks via Helm values, Longhorn cannot be managed in a fully GitOps / CI-CD driven workflow. Adding extra disks currently requires manual intervention through the UI, which breaks the declarative infrastructure model that Kubernetes is built around. For anyone running Longhorn with ArgoCD, Flux, or any other GitOps tooling, this is a gap that forces manual drift into otherwise automated environments.
@ejweber already outlined solid implementation approaches back in May 2024, and there seemed to be consensus among contributors that this fits well. Is there anything blocking this from getting scheduled again, or would a community contribution be welcome to move this forward?
